libvast/src/system/actor_system.cpp

```diff
@@ -0,0 +1,75 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/system/actor_system.hpp"
+
+#include "vast/config.hpp"
+#include "vast/defaults.hpp"
+#include "vast/logger.hpp"
+#include "vast/system/configuration.hpp"
+
+#include <caf/atom.hpp>
+
+namespace vast::system {
+
+namespace {
+
+/// Converts a verbosity atom to its integer counterpart.
+unsigned to_level_int(caf::atom_value x) {
+  switch (atom_uint(to_lowercase(x))) {
+    default:
+      return VAST_LOG_LEVEL_QUIET;
+    case caf::atom_uint("quiet"):
+      return VAST_LOG_LEVEL_QUIET;
+    case caf::atom_uint("error"):
+      return VAST_LOG_LEVEL_ERROR;
+    case caf::atom_uint("warning"):
+      return VAST_LOG_LEVEL_WARNING;
+    case caf::atom_uint("info"):
+      return VAST_LOG_LEVEL_INFO;
+    case caf::atom_uint("verbose"):
+      return VAST_LOG_LEVEL_VERBOSE;
+    case caf::atom_uint("debug"):
+      return VAST_LOG_LEVEL_DEBUG;
+    case caf::atom_uint("trace"):
+      return VAST_LOG_LEVEL_TRACE;
+  }
+}
+
+} // namespace
+
+void fixup_logger(const configuration& cfg) {
+  // Reset the logger so we can support the VERBOSE level
+  namespace lg = defaults::logger;
+  struct _hack : public caf::ref_counted {
+    caf::logger::config cfg_;
+  };
+  auto logger = caf::logger::current_logger();
+  auto& cfg_ = reinterpret_cast<_hack*>(logger)->cfg_;
```

Comment:
We should raise an issue in CAF and add a comment above this line with a link to the issue.

Reply:
@lava linked this as a better solution: http://bloglitb.blogspot.com/2010/07/access-to-private-members-thats-easy.html

I think this should be used instead to avoid possible alignment issues.

Reply:
Both done.